packages/stencil/components/modal/src/components/osds-modal/osds-modal.tsx

```diff
@@ -0,0 +1,117 @@
+import { Component, Element, Host, h, Prop, Method } from '@stencil/core';
+import {
+  OdsModal,
+  OdsModalController,
+  OdsModalEvents,
+  OdsModalMethods,
+  odsModalDefaultAttributes,
+  OdsIconSize,
+  OdsIconName,
+  OdsTextLevel,
+} from '@ovhcloud/ods-core';
+import { OdsStencilEvents, OdsStencilMethods } from '@ovhcloud/ods-stencil/libraries/stencil-core';
+import { OdsThemeColorIntent, OdsThemeTypographySize } from '@ovhcloud/ods-theming';
+
+/**
+ * @slot (unnamed) - Modal content
+ */
+@Component({
+  tag: 'osds-modal',
+  styleUrl: 'osds-modal.scss',
+  shadow: true
+})
+export class OsdsModal implements OdsModal<OdsStencilMethods<OdsModalMethods>, OdsStencilEvents<OdsModalEvents>> {
+  controller: OdsModalController = new OdsModalController(this);
+  @Element() el!: HTMLElement;
+
+  /** @see OdsModalAttributes.color */
+  @Prop({ reflect: true }) color: OdsThemeColorIntent = odsModalDefaultAttributes.color;
+
+  /** @see OdsModalAttributes.dismissible */
+  @Prop({ reflect: true }) dismissible?: boolean = odsModalDefaultAttributes.dismissible;
+
+  /** @see OdsModalAttributes.headline */
+  @Prop({ reflect: true }) headline?: string = odsModalDefaultAttributes.headline;
+
+  /** @see OdsModalAttributes.masked */
+  @Prop({ reflect: true, mutable: true }) masked?: boolean = odsModalDefaultAttributes.masked;
+
+  /**
+   * @see OdsModalMethods.close
+   */
+  @Method()
+  async close() {
+    this.controller.close();
+  }
+
+  /**
+   * @see OdsModalMethods.open
+   */
+  @Method()
+  async open() {
+    this.controller.open();
+  }
+
+  /**
+   * @see OdsModalBehavior.beforeRender
+   */
+  beforeRender(): void {
+
+  }
+
+  componentWillRender(): void {
+    document.documentElement.appendChild(this.el);
+  }
+
+  render() {
+
+    const {
+      color,
+      headline,
+      dismissible,
+      masked
+    } = this;
+
+    return (
+      <Host {...{ masked }}>
+        <div class="backdrop"></div>
+
+        <div class="wrapper">
+          <div class="header">
+            { dismissible &&
+              <osds-icon
```

Comment:
We can't focus this "button" with navigation key. I think for this you need to wrap the icon with a circle button